src/compiler/checker.ts

```diff
@@ -1813,25 +1813,35 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         isTypeParameterPossiblyReferenced,
     };
 
-    function runWithInferenceBlockedFromSourceNode<T>(node: Node | undefined, fn: () => T): T {
+    function runWithoutResolvedSignatureCaching<T>(node: Node | undefined, fn: () => T): T {
         const containingCall = findAncestor(node, isCallLikeExpression);

```

Comment:
this is duplicated work with runWithInferenceBlockedFromSourceNode, and if this function were inlined into getResolvedSignatureWorker, the node is already guaranteed to be a CallLikeExpression, no search required.

Reply:
Alternatively just pass in `containingCall` to `runWithoutResolvedSignatureCaching`.

Reply:
> this is duplicated work with runWithInferenceBlockedFromSourceNode, and if this function were inlined into getResolvedSignatureWorker, the node is already guaranteed to be a CallLikeExpression, no search required.

When called from `runWithInferenceBlockedFromSourceNode` - then yes. This is also now called directly from `getResolvedSignatureForSignatureHelp` as well and thus it's not that simple to reuse the existing computation (it's not worth it in my eyes).

Reply:
I hate doing the same thing twice but I agree that it's ideally a shallow walk that only happens in signature help.